From 2725f89965f8015c55ab2e57413ccecbe0d28c46 Mon Sep 17 00:00:00 2001
From: Xi Ruoyao <xry111@xry111.site>
Date: Mon, 23 Jun 2025 20:35:32 +0800
Subject: [PATCH 18/63] loonggpu: Disable outputs when releasing fbdev client

Following up the reference implementation change to avoid potential
errors.

Link: https://git.kernel.org/torvalds/c/c069dbbcba73
Signed-off-by: Xi Ruoyao <xry111@xry111.site>
---
 loonggpu/loonggpu_fb.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/loonggpu/loonggpu_fb.c b/loonggpu/loonggpu_fb.c
index e225499..a6707ff 100644
--- a/loonggpu/loonggpu_fb.c
+++ b/loonggpu/loonggpu_fb.c
@@ -286,6 +286,7 @@ static void loonggpu_fbdev_client_unregister(struct drm_client_dev *client)
 
 	if (fb_helper->info) {
 		vga_switcheroo_client_fb_set(rdev->pdev, NULL);
+		drm_helper_force_disable_all(dev);
 		drm_fb_helper_unregister_info(fb_helper);
 	} else {
 		drm_client_release(&fb_helper->client);
-- 
2.52.0

